Amazon.QLDB.Driver/driver/AsyncQldbDriver.cs

```diff
@@ -103,5 +230,51 @@ public async Task<IEnumerable<string>> ListTableNames(CancellationToken cancella
 
             return (await result.ToListAsync(cancellationToken)).Select(i => i.StringValue);
         }
+
+        private async Task<AsyncQldbSession> GetSession()
+        {
+            this.logger.LogDebug(
+                "Getting session. There are {} free sessions and {} available permits.",
+                this.sessionPool.Count,
+                this.sessionPool.BoundedCapacity - this.poolPermits.CurrentCount);
+
+            if (await this.poolPermits.WaitAsync(DefaultTimeoutInMs))
+            {
+                var session = this.sessionPool.Count > 0 ? this.sessionPool.Take() : null;
+
+                if (session == null)
+                {
+                    session = await this.StartNewSession();
+                    this.logger.LogDebug("Creating new pooled session with ID {}.", session.GetSessionId());
+                }
+
+                return session;
+            }
+            else
+            {
+                this.logger.LogError(ExceptionMessages.SessionPoolEmpty);
+                throw new QldbDriverException(ExceptionMessages.SessionPoolEmpty);
+            }
+        }
+
+        private async Task<AsyncQldbSession> StartNewSession()
+        {
+            try
+            {
+                Session session = await Session.StartSessionAsync(this.ledgerName, this.sessionClient, this.logger);
+                return new AsyncQldbSession(session, this.logger);
+            }
+            catch (Exception e)
+            {
+                throw new RetriableException("None", false, e);
```

Comment:
None what?

Reply:
Let me throw this into a constant variable.